airflow/api_fastapi/core_api/openapi/v1-generated.yaml

```diff
@@ -13,12 +13,16 @@ paths:
       - Asset
       summary: Next Run Assets
       operationId: next_run_assets
+      security:
+      - OAuth2PasswordBearer: []
       parameters:
       - name: dag_id
         in: path
         required: true
         schema:
-          type: string
+          anyOf:
+          - type: string
+          - type: 'null'
```

Comment:
This schema update is not normal.

That's because the permission decorator update the spec. The reason for that is because we to:
```
    def inner(
        user: Annotated[BaseUser, Depends(get_user)],
        dag_id: str | None = None,
    ) -> None:\
```

Similarly to other access function (check pool for example), we should 
```
    dag_id = request.path_params.get("dag_id") # instead of having the `dag_id` declared as a parameter.
```

It will solve the problem.

Reply:
just updated it. seems to update many other files as well 🤔